src/operator-manual/tracing/01-quickstart.md

```diff
@@ -0,0 +1,430 @@
+    the sidecar collector happens over gRPC.
+  * The collector will then export all the traces in batches. The traces will
+    be sent to a Jaeger collector.
+  * The Jaeger collector is exposed via the internal Service that was created
+    by the Jaeger Operator.
+  * The communication between the OpenTelemetry Collector and the Jaeger endpoint
+    is not secured by TLS. We are fine with this limitation because this is
+    just a demo environment.
+
+Everything is ready, we can now deploy Kubewarden.
+
+## Install Kubewarden
+
+We can proceed to the deployment of Kubewarden in the usual way.
+
+> **Note well:** cert-manager is a requirement of Kubewarden, but we've already
+> installed it before performing the deployment of the OpenTelemetry Operator.
+
+As a first step, we have to add the Helm repository that contains Kubewarden:
+
+```console
+helm repo add kubewarden https://charts.kubewarden.io
+```
+
+Then we have to install the Custom Resource Definitions (CRDs) defined by
+Kubewarden:
+
+```console
+helm install --create-namespace -n kubewarden kubewarden-crds kubewarden/kubewarden-crds
+```
+
+Now we can deploy the rest of the Kubewarden stack. The official
+helm chart will create a PolicyServer named `default`. We want this PolicyServer
+instance to have tracing enabled.
+
+In order to do that, we have to specify some extra values at installation time.
+Let's create a `values.yaml` file with the following contents:
+
+```yaml
+policyServer:
+  env:
+  - name: KUBEWARDEN_LOG_LEVEL
+    value: info
+  - name: KUBEWARDEN_LOG_FMT
+    value: otlp
+  annotations:
+    sidecar.opentelemetry.io/inject: "true"
+```
+
+Then we can proceed with the installation of the helm chart:
+
+```console
+helm install --wait --namespace kubewarden --values values.yaml kubewarden-controller kubewarden/kubewarden-controller
+```
+
+This leads to the creation of the `default` instance of `PolicyServer`:
+
+```console
+kubectl get -n kubewarden policyservers.policies.kubewarden.io -n kubewarden
+NAME      AGE
+default   3m7s
+```
+
+Looking closer at the Pod running the PolicyServer instance, we will find it has
+two containers inside of it: the actual `policy-server` and the OpenTelemetry
+Collector sidecar `otc-container`.
+
+## Enforcing a policy
+
+We will start by deploying the [safe-labels](https://github.com/kubewarden/safe-labels-policy)
+policy:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: ClusterAdmissionPolicy
+metadata:
+  name: safe-labels
+spec:
+  module: registry://ghcr.io/kubewarden/policies/safe-labels:v0.1.6
+  settings:
+    mandatory_labels:
+    - owner
+  rules:
+    - apiGroups:
+        - "*"
+      apiVersions:
+        - "*"
+      resources:
+        - "*"
+      operations:
+        - CREATE
+        - UPDATE
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values: ["default"]
+  mutating: false
+EOF
+```
```

Comment:
I cannot see any resource in https://github.com/derailed/k9s after applying this policy. It works fine with kubectl, so probably k9s is initializing resources without the owner label.  I don't think there is a problem with the policy itself, but for k9s users cluster seems to be broken as they cannot list any resource at all.
Error:
`│                           < Ruroh? admission webhook "safe-labels.kubewarden.admission" denied the request: The following mandatory labels are missing: owner >                            │
`

Reply:
I think each Kubernetes distro is doing its thing with automatic labels. I was running the original example against minikube, but it turns out k3s was not adding this label.

I've changed the example to use different key/values and create ad-hoc Namespace objects